internal/backend/runtime/omni/controllers/omni/machine_set_node.go

```diff
@@ -28,195 +30,246 @@ import (
 	"github.com/siderolabs/omni/client/pkg/omni/resources/system"
 )
 
+// Questions for pr
+// 1. which data is stored in the spec and which in the labels?
+
```

Comment:
This resource is used to extract the labels from any resource type.
In this case it's `omni.MachineStatus` labels.

As this controller doesn't use anything from the `omni.MachineStatus` spec, it can rely on the stripped down version of the resource.

Reply:
Okay I see, but is there a reason for why, for example, the evicted label shouldn't be an `isEvicted` field on the `omni.MachineStatus` spec?

Also, currently it seems that references to owner resources seem to be done using labels instead of spec fields. 
For example:
* machineSets have a `LabelCluster` instead of having a `clusterId` field in the spec
* machines resources have a `LabelMachineSet` label instead of a a `MachineSetId` or a similar field in the spec 

Is there a reason for that? Is it due to avoid having confusing zero values in the spec as the labels can be deleted?